python/lsst/ap/association/assoc_db_sqlite.py

```diff
@@ -0,0 +1,479 @@
+            name_type_string = name_type_string[:-1]
+            self._db_cursor.execute(
+                "CREATE TABLE dia_sources (%s)" % name_type_string)
+
+            self._db_cursor.execute(
+                "CREATE TABLE dia_objects_to_dia_soruces "
+                "(src_id INTEGER PRIMARY KEY, obj_id INTEGER)")
+
+            self._db_connection.commit()
+
+        return True
+
+    @pipeBase.timeMethod
+    def load(self, ctr_coord, radius):
+        """ Load all DIAObjects and associated DIASources.
+
+        Parameters
+        ----------
+        ctr_coord : lsst.afw.geom.Coord
+            Center position of on on sky circle to load.
+        radius : lsst.afw.geom.Angle
+
+
+        Returns
+        -------
+        lsst.ap.association.DIAObjectCollection
+        """
+        indexer_indices, on_boundry = self.indexer.get_pixel_ids(
+            ctr_coord, radius)
+
+        dia_objects = self.get_dia_objects(indexer_indices)
+
+        dia_collection = DIAObjectCollection(dia_objects)
+
+        return dia_collection
+
+    @pipeBase.timeMethod
+    def store(self, dia_collection, compute_spatial_index=False):
+        """ Store all DIAObjects and DIASources in this dia_collection.
+
+        Parameters
+        ----------
+        dia_collection : lsst.ap.association.DIAObjectCollection
+            Collection DIAObjects to store. Also stores the DIASources
+            associated with these DIAObjects.
+        compute_spatial_index : bool
+            If True, compute the spatial search indices using the
+            indexer specified at class creation.
+        """
+        for dia_object in dia_collection.dia_objects:
+            if compute_spatial_index:
+                dia_object.dia_object_record.set(
+                    'indexer_id', self.indexer.index_points(
+                        [dia_object.ra], [dia_object.dec])[0])
+            self.store_dia_object(dia_object)
+            for dia_source in dia_object.dia_source_catalog:
+                self.store_dia_source(dia_source)
+                self.store_dia_object_source_pair(
+                    dia_object.get('id'), dia_source.getId())
+        self.commit()
+
+    @pipeBase.timeMethod
+    def store_updated(self, dia_collection, updated_indices):
+        """ Store new DIAObjects and sources in the sqlite database.
+
+        Parameters
+        ----------
+        dia_collection : lsst.ap.association.DIAObjectCollection
+        updated_indices : int ndarray
+            Indices within the DIAObjctCollection that contain DIAObjects
+            that were updated by AssociationTask.
+        """
+        for updated_index in updated_indices:
+            dia_object = dia_collection.dia_objects[updated_index]
+            if dia_object.n_dia_sources == 1:
+                dia_object.dia_object_record.set(
+                    'indexer_id', self.indexer.index_points(
+                        [dia_object.ra], [dia_object.dec])[0])
```

Comment:
This is an update in case the updated DIAObject, with new DIASources has shifted from one indexer cell to the next, or it is a newly created DIAObject. As above, the index here refers to the indexer where as updated_index refers to the index in the set of DIAObjects. See my previous comments.

Reply:
The problem is actually worse with a list index than a spatial index -- as soon as you change one index, all values in `updated_indices` will point to the wrong object.